src/Translation/Translator.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\Translation;
+
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Translator.
+ *
+ * @author Martin Auswöger <martin@auswoeger.com>
+ */
+class Translator implements TranslatorInterface
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var ContaoFrameworkInterface
+     */
+    private $framework;
+
+    /**
+     * @param TranslatorInterface      $translator Original translator service that gets decorated.
+     * @param ContaoFrameworkInterface $framework
+     */
+    public function __construct(TranslatorInterface $translator, ContaoFrameworkInterface $framework)
+    {
+        $this->translator = $translator;
+        $this->framework = $framework;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function trans($id, array $parameters = [], $domain = null, $locale = null)
+    {
+        if (null === $domain || strncmp($domain, 'contao_', 7) !== 0) {
+            return $this->translator->trans($id, $parameters, $domain, $locale);
+        }
+
+        if (!$this->framework->isInitialized()) {
```

Comment:
Should we initialize the framework here?

Reply:
Yes. Done.